### DISCUSSION TITLE
Does wayvnc/neatvnc support Unicode text in the clipboard pasting between the client and server?

### DISCUSSION THREAD

**Discussion by layercak3** [any1/wayvnc#134] [Q&A] [answered]
Hi, I'm using wayvnc 0.4.1, neatvnc 0.4.0, and latest sway & wlroots from git master on an Arch Linux computer. On another Arch Linux computer I'm using tigervnc 1.12.0 as the client, and latest sway & wlroots from git master. When pasting Unicode characters from the client clipboard to the server, they appear as "?". When I paste Unicode characters from the server clipboard to the client, I get garbled text. For example, "μ" > "Î¼". Copying and from the server and pasting it on the server works, just not between the client and server. I believe the tigervnc client has support for it now, but I'm not sure if wayvnc/neatvnc does so I would like to know if the problem lies in wayvnc/neatvnc. My keyboard layout on the client is the same as the one set in wayvnc.

**Comment by any1** [comment]
The standard specifies ISO 8859-1 only. See https://github.com/rfbproto/rfbproto/blob/master/rfbproto.rst#servercuttext. There is an extension which specifies UTF-8, but it's not implemented. See: https://github.com/rfbproto/rfbproto/blob/master/rfbproto.rst#extended-clipboard-pseudo-encoding.

**Comment by layercak3** [comment]
For now I will just copy text into a file from ssh and have a terminal window in the VNC window then do `wl-copy < pseudo-clipboard`, or copy text into a file from a terminal in the VNC window then copy the text from the ssh window XD

**Comment by layercak3** [comment, marked as answer]
Yes it does, if you update to 0.9.0 and have a compatible client
